torch/_dynamo/allowed_functions.py

```diff
@@ -148,16 +148,37 @@ def _disallowed_function_ids() -> Set[int]:
     return {id(x) for x in remove}
 
 
-# We are in progress of refactoring and moving the following functions to test_trace_rules.py.
-# If you made any change to the following functions, please also update there as well.
-# If you are not clear of how to update, please contact @yanboliang.
-@FunctionIdSet
-def _allowed_function_ids() -> Dict[int, str]:
+# Helper function to dump the torch name rule map generated based on
+# the heuristic defined in gen_allowed_objs_and_ids.
+def dump_allowed_torch_name_rule_map() -> None:
+    m = gen_allowed_objs_and_ids()[2]
+    for k, v in m.items():
+        print(f'"{k}": {v.__name__},')
+
+
+def gen_allowed_objs_and_ids() -> Tuple[Dict[int, str], Set[Any], Dict[str, Any]]:
```

Comment:
Return a NamedTuple instead of a tuple

Reply:
or better yet, a dataclass, to forbid positional access

Reply:
SG!

Reply:
Resolved!